vendors/st/boards/stm32l475_discovery/ports/common_io/iot_i2c.c

```diff
@@ -0,0 +1,744 @@
+
+            break;
+
+        case eI2CGetBusState:
+
+            if( pvBuffer == NULL )
+            {
+                return IOT_I2C_INVALID_VALUE;
+            }
+
+            *( IotI2CBusStatus_t * ) pvBuffer = pI2cDescriptor->xHandle.State == HAL_I2C_STATE_READY ? eI2CBusIdle : eI2cBusBusy;
+
+            break;
+
+        /* TODO: have not tested */
+        case eI2CBusReset:
+
+            DISCOVERY_I2Cx_FORCE_RESET();
+
+            DISCOVERY_I2Cx_RELEASE_RESET();
+
+            break;
+
+        case eI2CGetTxNoOfbytes:
+
+            if( pvBuffer == NULL )
+            {
+                return IOT_I2C_INVALID_VALUE;
+            }
+
+            *( uint16_t * ) pvBuffer = pxI2CPeripheral->usTransmittedTxBytes;
+
+            break;
+
+        case eI2CGetRxNoOfbytes:
+
+            if( pvBuffer == NULL )
+            {
+                return IOT_I2C_INVALID_VALUE;
+            }
+
+            *( uint16_t * ) pvBuffer = pxI2CPeripheral->usReceivedRxBytes;
+
+            break;
+
+        default:
+            status = IOT_I2C_INVALID_VALUE;
+    }
+
+    return status;
+}
+
+int32_t iot_i2c_close( IotI2CHandle_t const pxI2CPeripheral )
+{
+    IotI2CDescriptor_t * pI2cDescriptor = pxI2CPeripheral;
+
+    if( ( pI2cDescriptor == NULL ) ||
+        ( ( pI2cDescriptor->flags & _FLAG_OPENED ) == 0 ) )
+    {
+        return IOT_I2C_INVALID_VALUE;
+    }
+
+    HAL_I2C_DeInit( &pI2cDescriptor->xHandle );
+
+    pI2cDescriptor->xCallback = NULL;
+    pI2cDescriptor->pvUserContext = NULL;
+    pI2cDescriptor->usSlaveAddr = _UNSET_SLAVE_ADDRESS;
+    pI2cDescriptor->usTransmittedTxBytes = 0;
+    pI2cDescriptor->usReceivedRxBytes = 0;
+    pI2cDescriptor->flags = _FLAG_INITIALIZER;
+    pI2cDescriptor->xConfig.ulBusFreq = 0;
+    pI2cDescriptor->xConfig.ulMasterTimeout = 0;
+
+    vSemaphoreDelete( pI2cDescriptor->xSemphr );
+
+    /* HAL_I2C_DeInit always returns HAL_OK as long as the handle is not NULL. */
+    return IOT_I2C_SUCCESS;
+}
+
+/*-----------------------------------------------------------*/
+
+/*
+ * Actual ISR registered in vector table.
+ */
+void I2C1_EV_IRQHandler( void )
+{
+    HAL_I2C_EV_IRQHandler( &( _i2cContexts[ 0 ].xHandle ) );
+}
+
+void I2C1_ER_IRQHandler( void )
+{
+    HAL_I2C_ER_IRQHandler( &( _i2cContexts[ 0 ].xHandle ) );
+}
+
+void I2C2_EV_IRQHandler( void )
+{
+    HAL_I2C_EV_IRQHandler( &( _i2cContexts[ 1 ].xHandle ) );
+}
+
+void I2C2_ER_IRQHandler( void )
+{
+    HAL_I2C_ER_IRQHandler( &( _i2cContexts[ 1 ].xHandle ) );
+}
+
+void I2C3_EV_IRQHandler( void )
+{
+    HAL_I2C_EV_IRQHandler( &( _i2cContexts[ 2 ].xHandle ) );
+}
+
+void I2C3_ER_IRQHandler( void )
+{
+    HAL_I2C_ER_IRQHandler( &( _i2cContexts[ 2 ].xHandle ) );
+}
+
+/*-------------ST API Implementation Overrides---------------*/
+
+/**
+ * @brief Initializes low level hardware
+ */
+void HAL_I2C_MspInit( I2C_HandleTypeDef * hi2c )
+{
+    GPIO_InitTypeDef xGpio_init_structure;
+
+    /* Configure the GPIOs */
```

Comment:
Coding style calls for comments to be complete sentences.

Reply:
Agree. I can say "Enable GPIO clock". Other than that, I don't more knowledge about this.